lib/workers/repository/update/pr/changelog/release-notes.ts

```diff
@@ -166,10 +173,13 @@ export async function getReleaseNotes(
 
 function getExactReleaseMatch(
   packageName: string,
+  depName: string,
   version: string,
   releases: ChangeLogNotes[],
 ): ChangeLogNotes | undefined {
-  const exactReleaseReg = regEx(`${packageName}[@_-]v?${version}`);
+  const exactReleaseReg = regEx(
+    `(?:${depName}|${packageName})[@_-]v?${version}$`,
```

Comment:
> [!NOTE]
> Terminated the regex with `$` to make it "more exact" as a scout habit; feel free to revert if desired.

Reply:
Best to revert this because we don't know the impact, plus we are more likely to want "loose" matching than to be extra strict

Reply:
Following [my comment below](https://github.com/renovatebot/renovate/pull/30395/files#r1692724285), I'd like to further improve this to avoid unintentional partial matches (while still allowing for tag prefixes):
```suggestion
    `(?:^|\/)(?:${depName}|${packageName})[@_-]v?${version}$`,
```

Reply:
> Best to revert this because we don't know the impact, plus we are more likely to want "loose" matching than to be extra strict

After some code inspection, it looks like there's no impact here since the first filter to collect `candidateReleases` already ensures that the corresponding tag will strictly end with the `version`. Similarly, in the context of `ChangeLogSource`'s `findTagOfRelease()`, there's an equality check in the last filter to ensure the version matches exactly. So no problem to revert the change to those regex patterns. 🙂

I'm still interested to add the prefix referenced in my pervious comment as part of this PR in order to reduce the chance of partial matching issues. WDYT?